src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobWorkitemStore.cs

```diff
@@ -58,12 +58,13 @@ public class BlobWorkitemStore : IWorkitemStore
         public async Task AddWorkitemAsync(
             WorkitemInstanceIdentifier identifier,
             DicomDataset dataset,
+            long? proposedWatermark = default,
```

Comment:
Justing checking, why cant we pass the proposed watermark in the identifier itself to keep it simple.

Reply:
I hesitantly added Watermark in the identifier. "Proposed" is not part of the identifier "yes". Hence avoided it.

Reply:
No worries, discussed with Josiah. fine with this approach